src/components/Script.jsx

```diff
@@ -20,14 +22,14 @@ const Script = ({ actors, currentScript }) => {
   const lineChunks = [];
   const characterSet = new Set(currentCharacters);
 
-  for (let i = 0; i < script.length; i++) {
+  for (let lineChunk of script) {
     // checks if the character name for this chunk is the name of a character assigned to the current actor
     // remove the dot because some scripts have a dot after the name of the character, ie VIOLA.
-    const name = script[i][0].replace('.', '');
+    const name = lineChunk.split('.')[0];
     if (characterSet.has(name)) {
-      lineChunks.push(<pre className='currentActor'>{script[i].join('\n')}</pre>);
+      lineChunks.push(<pre className='currentActor'>{lineChunk}</pre>);
```

Comment:
parsing scripts now returns one string for each line chunk, instead of an array of strings

Reply:
Nice simplification.